### ISSUE TITLE
Some syntax issues with gcc-14 and treating warnings as error

### ISSUE THREAD

**Issue by tillea** [n-t-roff/sc#33]
Hi,
I'm just upgrading the Debian packaged sc to your Github version.  When doing so I needed to apply some patches [which you can see in our development platform](https://salsa.debian.org/debian/sc/-/blob/master/debian/patches/syntax.patch?ref_type=heads).  Feel free to take these patches over in your next release.
Kind regards,   Andreas.

**Comment by dtkerns** [comment]
All these look like good changes, but alas, the only one that looks like it might still apply is this one:
```diff
--- a/cmds.c
+++ b/cmds.c
@@ -2997,7 +2997,7 @@ readfile(char *fname, int eraseflg)
 		return 0;
 	    }
 	    *save = '|';
-	    if ((strlen(save) + strlen(fname) + 2) > PATHLEN) {
+	    if ((strlen(save) + strlen(fname) + 4) > PATHLEN) {
 		error("Path too long");
 		return 0;
 	    }
```
Are you sure you're current?
btw, I just pushed a warning fix reported by gcc (GCC) 11.4.1 20231218

**Comment by dtkerns** [comment]
see Commit 93cca02

**Comment by tillea** [comment]
Am Thu, Nov 21, 2024 at 09:24:25PM -0800 schrieb dtkerns:
> Are you sure you're current?

Debian packaging usually fetches the latest tag not latest commit.  If
these patches are applied it would be really great if you could push a
new tag.

Thanks a lot for your quick reply, Andreas.

**Comment by dtkerns** [comment]
@n-t-roff what's your process for updating the version string for a new tag?

**Comment by tillea** [comment]
There is a program in Debian (uscan) which checks upstream for a new regexp matching the version.  On Github that's simply the tag.  Currently it results in:

    uscan info: Matching pattern:
       (?:(?:https://github.com)?\/n\-t\-roff\/sc\/)?(?:.*?/)?v?(\d[\d._]*)\.tar\.gz
    uscan info: Found the following matching hrefs on the web page (newest first):
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.2.tar.gz (7.16-1.1.2) index=7.16-1.1.2-1
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.2.tar.gz (7.16-1.1.2) index=7.16-1.1.2-1
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.1.tar.gz (7.16-1.1.1) index=7.16-1.1.1-1
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.1.tar.gz (7.16-1.1.1) index=7.16-1.1.1-1
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.0.tar.gz (7.16-1.1.0) index=7.16-1.1.0-1
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.0.tar.gz (7.16-1.1.0) index=7.16-1.1.0-1
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.0.1.tar.gz (7.16-1.0.1) index=7.16-1.0.1-1
       https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.0.1.tar.gz (7.16-1.0.1) index=7.16-1.0.1-1
    uscan info: Looking at $base = https://github.com/n-t-roff/sc/tags with
        $filepattern = (?:.*?/)?v?(\d[\d._]*)\.tar\.gz found
        $newfile     = https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.2.tar.gz
        $newversion  = 7.16-1.1.2
        $lastversion = 7.16-1.1.2
    uscan info: Matching target for downloadurlmangle: https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.2.tar.gz
    uscan info: Upstream URL(+tag) to download is identified as    https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.2.tar.gz
    uscan info: Matching target for filenamemangle: https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.2.tar.gz
    uscan info: Filename (filenamemangled) for downloaded file: sc-1.1.2.tar.gz
    uscan info: Newest version of sc on remote site is 7.16-1.1.2, local version is 7.16-1.1.2
    uscan info:  => Package is up to date from:
                 => https://github.com/n-t-roff/sc/archive/refs/tags/7.16_1.1.2.tar.gz

**Comment by dtkerns** [comment]
ok. I have the version thing figured out... I'd really like to include a fix for #21 in this version; @tillea any chance I get get you to test the 1-line change I put in that thread? would be even better if you knew the correct place to do that initialization!

**Comment by tillea** [comment]
Am Fri, Nov 22, 2024 at 08:31:15PM -0800 schrieb dtkerns:
> ok. I have the version thing figured out... I'd really like to include a fix for #21 in this version; @tillea any chance I get get you to test the 1-line change I put in that thread? would be even better if you knew the correct place to do that initialization!

BTW, I would give up on the 7.16-my.own.version and continue with
7.17 or even 8.0

**Comment by dtkerns** [comment]
I'll do that after I solve #21 :)
